patches/server/1045-Vault-API.patch

```diff
@@ -0,0 +1,278 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Boy <sivertpaulsen2@gmail.com>
+Date: Wed, 24 Jul 2024 13:53:58 +0200
+Subject: [PATCH] Vault API
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/entity/vault/VaultServerData.java b/src/main/java/net/minecraft/world/level/block/entity/vault/VaultServerData.java
+index 30544783d959fbdf4b6fd53c10bd3806e718a066..5b245cca7c9e7a141b2c1b7f145930b67bf75171 100644
+--- a/src/main/java/net/minecraft/world/level/block/entity/vault/VaultServerData.java
++++ b/src/main/java/net/minecraft/world/level/block/entity/vault/VaultServerData.java
+@@ -19,12 +19,12 @@ public class VaultServerData {
+     static final String TAG_NAME = "server_data";
+     static Codec<VaultServerData> CODEC = RecordCodecBuilder.create(
+         instance -> instance.group(
+-                    UUIDUtil.CODEC_LINKED_SET.lenientOptionalFieldOf("rewarded_players", Set.of()).forGetter(data -> data.rewardedPlayers),
+-                    Codec.LONG.lenientOptionalFieldOf("state_updating_resumes_at", Long.valueOf(0L)).forGetter(data -> data.stateUpdatingResumesAt),
+-                    ItemStack.CODEC.listOf().lenientOptionalFieldOf("items_to_eject", List.of()).forGetter(data -> data.itemsToEject),
+-                    Codec.INT.lenientOptionalFieldOf("total_ejections_needed", Integer.valueOf(0)).forGetter(data -> data.totalEjectionsNeeded)
+-                )
+-                .apply(instance, VaultServerData::new)
++                UUIDUtil.CODEC_LINKED_SET.lenientOptionalFieldOf("rewarded_players", Set.of()).forGetter(data -> data.rewardedPlayers),
++                Codec.LONG.lenientOptionalFieldOf("state_updating_resumes_at", Long.valueOf(0L)).forGetter(data -> data.stateUpdatingResumesAt),
++                ItemStack.CODEC.listOf().lenientOptionalFieldOf("items_to_eject", List.of()).forGetter(data -> data.itemsToEject),
++                Codec.INT.lenientOptionalFieldOf("total_ejections_needed", Integer.valueOf(0)).forGetter(data -> data.totalEjectionsNeeded)
++            )
++            .apply(instance, VaultServerData::new)
```

Comment:
I don't think these indentations should be changed here as this just adds more potential conflicts for future updates

Reply:
Yeah not to sure why this is being weird...
Saw it originally though didnt indent it myself

Recent commit here is directly copied from the original file and should therefore not touch the indentation, though seems it does

Reply:
You have to be weary of where the "original file" comes from, we use a different decompiler to spigot which tends to produce a more stable (but different) output vs spigots decompiler. if spigot touched this class, it will be handled by their decompiler

Reply:
should be fixed now, it and the few other weird unintended diffs 👍